File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/persistence/MovieDAO.java
package com.github.rahmnathan.localmovies.app.persistence;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface MovieDAO {
    @Query("SELECT * FROM MovieEntity")
    List<com.github.rahmnathan.localmovies.app.persistence.MovieEntity> getAll();

    @Query("select * from MovieEntity where directoryPath = :path and filename = :name limit 1")
    com.github.rahmnathan.localmovies.app.persistence.MovieEntity getByPathAndFilename(String path, String name);

    @Insert
    void insertAll(List<MovieEntity> movieEntities);

    @Insert
    void insert(MovieEntity movieEntity);

    @Query("delete from MovieEntity")
    void deleteAll();

    @Delete
    void delete(com.github.rahmnathan.localmovies.app.persistence.MovieEntity movie);
}

File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/persistence/MovieDAO.kt
package com.github.rahmnathan.localmovies.app.persistence

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface MovieDAO {
    @get:Query("SELECT * FROM MovieEntity")
    val all: List<MovieEntity?>?

    @Query("select * from MovieEntity where directoryPath = :path and filename = :name limit 1")
    fun getByPathAndFilename(path: String?, name: String?): MovieEntity?

    @Insert
    fun insertAll(movieEntities: List<MovieEntity?>?)

    @Insert
    fun insert(movieEntity: MovieEntity?)

    @Query("delete from MovieEntity")
    fun deleteAll()

    @Delete
    fun delete(movie: MovieEntity?)
}